diff -crB cyrus-sasl.original/include/sasl/sasl.h cyrus-sasl/include/sasl/sasl.h
*** cyrus-sasl.original/include/sasl/sasl.h	2010-11-25 17:48:04.000000000 +0100
--- cyrus-sasl/include/sasl/sasl.h	2010-11-25 17:50:40.000000000 +0100
***************
*** 346,352 ****
       * Mechanisms must ignore callbacks with id's they don't recognize.
       */
      unsigned long id;
!     int (*proc)();   /* Callback function.  Types of arguments vary by 'id' */
      void *context;
  } sasl_callback_t;
  
--- 346,352 ----
       * Mechanisms must ignore callbacks with id's they don't recognize.
       */
      unsigned long id;
!     int (*proc);   /* Callback function.  Types of arguments vary by 'id' */
      void *context;
  } sasl_callback_t;
  
