--- java_libs/com/lowagie/text/ImgPostscript.java
+++ java_libs/com/lowagie/text/ImgPostscript.java
@@ -53,7 +53,6 @@
 import java.io.*;
 import java.net.*;
 import com.lowagie.text.pdf.*;
-import com.lowagie.text.pdf.codec.postscript.*;
 import java.util.StringTokenizer;
 
 /**
@@ -194,7 +193,7 @@
      * @throws IOException on error
      * @throws DocumentException on error
      */
-    public void readPostscript(PdfTemplate template) throws IOException,
+/*    public void readPostscript(PdfTemplate template) throws IOException,
     DocumentException {
         setTemplateData(template);
         template.setWidth(width());
@@ -215,5 +214,5 @@
                 is.close();
             }
         }
-    }
+    } */
 }
--- java_libs/com/lowagie/text/pdf/PdfWriter.java
+++ java_libs/com/lowagie/text/pdf/PdfWriter.java
@@ -947,7 +947,7 @@
                         catch (Exception e) {
                             throw new DocumentException(e);
                         }
-                    }else{
+                    } /* else{
                         try {
                             ((ImgPostscript)image).readPostscript(getDirectContent().createTemplate(0, 0));
                         }
@@ -955,7 +955,7 @@
                             throw new DocumentException(e);
                         }
                         
-                    }
+                    } */
                 }
             }
             else {
