--- java_libs/Makefile.orig	2007-08-17 10:31:29.000000000 +0200
+++ java_libs/Makefile	2007-08-17 10:17:09.000000000 +0200
@@ -33,7 +33,7 @@
 	$(MAKE) -C "$(java_libs_root)/com/lowagie/text/pdf/fonts";
 	$(MAKE) -C "$(java_libs_root)/com/lowagie/text/xml/xmp";
 	$(MAKE) -C "$(java_libs_root)/com/lowagie/bc/asn1";
-	$(MAKE) -C "$(java_libs_root)/com/lowagie/text/pdf/codec/postscript";
+#	$(MAKE) -C "$(java_libs_root)/com/lowagie/text/pdf/codec/postscript";
 
 clean : libgcj_local_clean itext_clean
 
@@ -55,4 +55,4 @@
 	$(MAKE) -iC "$(java_libs_root)/com/lowagie/text/pdf/fonts" clean;
 	$(MAKE) -iC "$(java_libs_root)/com/lowagie/text/xml/xmp" clean;
 	$(MAKE) -iC "$(java_libs_root)/com/lowagie/bc/asn1" clean;
-	$(MAKE) -iC "$(java_libs_root)/com/lowagie/text/pdf/codec/postscript" clean;
+#	$(MAKE) -iC "$(java_libs_root)/com/lowagie/text/pdf/codec/postscript" clean;
--- pdftk/Makefile.Base.orig	2007-08-17 10:33:21.000000000 +0200
+++ pdftk/Makefile.Base	2007-08-17 10:15:57.000000000 +0200
@@ -25,7 +25,8 @@
 $(java_libs_root)/com/lowagie/text/markup/markup.a \
 $(java_libs_root)/com/lowagie/text/xml/xmp/xmp.a \
 $(java_libs_root)/com/lowagie/bc/asn1/asn1.a \
-$(java_libs_root)/com/lowagie/text/pdf/codec/postscript/postscript.a \
+
+# $(java_libs_root)/com/lowagie/text/pdf/codec/postscript/postscript.a \
 
 # font metric files, included as resources
 afms= $(wildcard $(java_libs_root)/com/lowagie/text/pdf/fonts/*.afm)
