--- pdftk/Makefile.Base.orig	2010-09-30 00:46:54.051606541 +0200
+++ pdftk/Makefile.Base	2010-09-30 00:51:06.887356676 +0200
@@ -55,7 +55,8 @@
 	$(CXX) pdftk.o attachments.o report.o $(itext_libs) $(libgcj_local_libs) $(afm_objects) -I$(java_libs_root) $(CPPFLAGS) $(CXXFLAGS) -o pdftk
 
 install:
-	/usr/bin/install pdftk /usr/local/bin 
+	mkdir -p ${DEST}/bin
+	/usr/bin/install pdftk ${DEST}/bin
 
 uninstall:
 	rm -vf /usr/local/bin/pdftk
