--- /home/luke/work/P-SAFI/pdftk/pdftk-1.41/pdftk/Makefile.RedHat	2006-09-21 19:37:38.000000000 +0200
+++ pdftk/Makefile.RedHat	2010-10-01 14:50:10.000000000 +0200
@@ -26,7 +26,7 @@
 # if you want pdftk to ask before overwriting a file, set
 # ASK_ABOUT_WARNINGS to true; otherwise: false; override this default
 # with the dont_ask or do_ask command-line options
-CPPFLAGS= -O2 -DPATH_DELIM=0x2f -DASK_ABOUT_WARNINGS=false -fdollars-in-identifiers
+CPPFLAGS= -O2 -DPATH_DELIM=0x2f -DASK_ABOUT_WARNINGS=false -fdollars-in-identifiers -I/usr/include/libgcj-4.4.3/
 CXXFLAGS= -lgcj
 
 # itext compiler flags
