---
 pdftk/pdftk.cc |    4 ++++
 1 file changed, 4 insertions(+)

--- a/pdftk/pdftk.cc
+++ b/pdftk/pdftk.cc
@@ -71,6 +71,8 @@
 #include "com/lowagie/text/pdf/PdfIndirectObject.h"
 #include "com/lowagie/text/pdf/PdfFileSpecification.h"
 #include "com/lowagie/text/pdf/PdfBoolean.h"
+#include "com/lowagie/text/pdf/PdfEncodings.h"
+#include "com/lowagie/text/pdf/BaseFont.h"
 
 #include "com/lowagie/text/pdf/RandomAccessFileOrArray.h" // for InputStreamToArray()
 
@@ -2672,6 +2674,8 @@ int main(int argc, char** argv)
 			JvInitClass(&itext::PdfDictionary::class$);
 			JvInitClass(&itext::PdfOutline::class$);
 			JvInitClass(&itext::PdfBoolean::class$);
+			JvInitClass(&itext::PdfEncodings::class$);
+			JvInitClass(&itext::BaseFont::class$);
 
 			TK_Session tk_session( argc, argv );
 
