--- setup.py.orig	2007-02-04 03:04:59.000000000 +0100
+++ setup.py	2007-02-04 03:09:46.000000000 +0100
@@ -684,7 +684,7 @@
                   and find_file("gdbm/ndbm.h", inc_dirs, []) is not None):
                 exts.append( Extension('dbm', ['dbmmodule.c'],
                                        define_macros=[('HAVE_GDBM_NDBM_H',None)],
-                                       libraries = ['gdbm'] ) )
+                                       libraries = ['gdbm_compat', 'gdbm'] ) )
             elif db_incs is not None:
                 exts.append( Extension('dbm', ['dbmmodule.c'],
                                        library_dirs=dblib_dir,
