--- cyrus-sasl-2.1.23/include/sasl.h	2010-11-25 18:15:05.000000000 +0100
+++ cyrus-sasl-2.1.23/include/sasl.h	2010-11-25 18:15:34.000000000 +0100
@@ -346,7 +346,7 @@
      * Mechanisms must ignore callbacks with id's they don't recognize.
      */
     unsigned long id;
-    int (*proc)();   /* Callback function.  Types of arguments vary by 'id' */
+    int (*proc);   /* Callback function.  Types of arguments vary by 'id' */
     void *context;
 } sasl_callback_t;
 
