--- Products/CMFCore/ActionsTool.py
+++ Products/CMFCore/ActionsTool.py
@@ -159,7 +159,7 @@
                 actions.extend( self._listActionInfos(provider, object) )
 
         # Include actions from object.
-        if object is not None:
+        if 0: #object is not None:
             base = aq_base(object)
             if IActionProvider.isImplementedBy(base):
                 actions.extend( object.listActionInfos(object=object) )
